### ISSUE TITLE
Suggesting to use `sortOn` when it doesn't exist for this type

### ISSUE THREAD

**Issue by JivanRoquet** [ndmitchell/hlint#1241]
Assuming the following imports:

```haskell
import           Data.Ord
import qualified Data.Vector                  as V
import           Data.Vector.Algorithms.Intro (sortBy)
```

Given the following code:

```haskell
V.modify (sortBy $ flip $ comparing fst) myVector
```

HLint suggests to do this instead:

```haskell
V.modify (sortOn (Down . fst)) myVector
```

The problem is that `sortOn` is not available for Vector (even with vector-algorithms) and results in a `Variable not in scope: sortOn` error.

I'm not sure if this is a bug or if this is expected behaviour, and also if there is a way to either silence this suggestion.

**Comment by ndmitchell** [comment]
It's not desirable behaviour, but it is somewhat expected. We should be noticing that the `sortBy` isn't `Data.List.sortBy`, but actually a vector one, and thus not matching. #157 is the first ticket that covers the same underlying issue, but we're now collecting them with a tag, so I'll add this one.

Unfortunately the only way to get rid of this hint is just silence it entirely, as per https://github.com/ndmitchell/hlint/blob/master/README.md#ignoring-hints

**Comment by ndmitchell** [comment]
I'd say if the ordering function uses a lot of memory you definitely want to avoid recreating that memory a lot, and potentially invoking the GC more than necessary. The case where `sortBy` is more efficient is usually when the function is a projection, just grabbing an already-computed field in the type. That said, if there were benchmarks showing the difference, we could adjust HLint in light of those.
